config/config.go

```diff
@@ -103,10 +104,26 @@ func loadForFileFromBytes(confBytes []byte, filePath string, kmsEncryptionContex
 	var rule *creationRule
 
 	for _, r := range conf.CreationRules {
-		if match, _ := regexp.MatchString(r.FilenameRegex, filePath); match {
+		if r.PathRegex == "" && r.FilenameRegex == "" {
 			rule = &r
 			break
 		}
+		if r.PathRegex != "" && r.FilenameRegex != "" {
+			return nil, fmt.Errorf("error loading config: both filename_regex and path_regex were found, use only path_regex")
+		}
+		if r.FilenameRegex != "" {
+			if match, _ := regexp.MatchString(r.FilenameRegex, filePath); match {
```

Comment:
I'm on the fence as to whether we should fix `filename_regex` and make it match on just the file name instead of the compete path or leaving it as it is. In the Python version `filename_regex` matched on just the file name, so technically this is a bug introduced in the Go version, which matches on the whole path. Thoughts?

cc @jvehent @ajvb

Reply:
We're personally using it to match on the full path and it's amazing! That would make us sad if that went away.

Reply:
I'm obviously quite new to sops, so interested to hear what @jvehent thinks, but the fact that both @devinburnette and those in the discussion in #77 are taking advantage of the bug pushes me towards voting for this PR. Keep it how it is as it is a feature users want, but move towards a more accurate naming.

Reply:
![mandatory xkcd](https://imgs.xkcd.com/comics/workflow.png)

Reply:
With that said, I don't think we should change the current behavior of `filename_regex`. Matching on the full path is a useful feature.